corehq/ex-submodules/phonelog/utils.py

```diff
@@ -277,3 +280,24 @@ def force_close_subreport(self):
                 device_model=log.get('device_model'),
             ) for log in logs
         ).encode('utf-8'))
+
+
+def clear_device_log_request(domain, xform):
```

Comment:
@shubham1g5 could I get your take on this logic, please?  A lot of the device log logic pulls username from one of the logs, or from a possible user registration entry at the top of the report.  Here I'm using the user ID of the user who submitted the logs - would you anticipate that these forced requests might not be authenticated in some cases, or might come from users that don't exist on HQ?

Reply:
Great question. I don't think the 2 user ids should be different in any current scenarios and both of them are reliably available, though if there is not any difference between the effort to use one vs other,  it makes more sense to me to use the one from the  `user_subreport` section so that in future we can reliably support any back door mechanism to send this log file to HQ for ex. things like transferring the file to a computer and uploading it to HQ. I would enforce though that features like these are very unlikely and we should prioritize 1. effort and 2. code consistency to anything else to make this decision. 

> Is device_id expected to be reliably present in the log data (like this)?
I didn't realize `device_id` gets written into the file but yeah it's reliable and always there. I think we should use this instead of extracting it from request params as originally planned. Let me know if you agree and I can revert mobile changes to include device_id as a query param for log submission requests.

Reply:
> Let me know if you agree and I can revert mobile changes to include device_id as a query param for log submission requests.

Yeah, that sounds good to me.  I just made a change to use the `user_subreport` if available, otherwise falling back to the user_id of the submitting user.

Reply:
Awesome 💯